tests/test_MultiProcDataset.py

```diff
@@ -64,6 +64,22 @@ def test_MultiProcDataset_n3_b5_shuffle():
         compare_dataset_seqs(hdf_dataset_seqs, mp_dataset_seqs)
 
 
+def test_MultiProcDataset_n3_b5_shuffle_sharding():
+    hdf_fn = generate_hdf_from_other({"class": "Task12AXDataset", "num_seqs": 23})
+    hdf_dataset_dict = {"class": "HDFDataset", "files": [hdf_fn], "seq_ordering": "random"}
+    hdf_dataset = init_dataset(hdf_dataset_dict)
+    hdf_dataset_seqs = dummy_iter_dataset(hdf_dataset)
+
+    with timeout():
+        mp_dataset = MultiProcDataset(
+            dataset=hdf_dataset_dict, num_workers=3, buffer_size=5, sharding_method="dedicated"
+        )
+        mp_dataset.initialize()
+        mp_dataset_seqs = dummy_iter_dataset(mp_dataset)
+
+        compare_dataset_seqs(hdf_dataset_seqs, mp_dataset_seqs)
```

Comment:
This test is wrong. `compare_dataset_seqs` assumes same order. But with `MultiProcDataset` and `sharding_method="dedicated"`, you will not have the same order.